src/window.cpp

```diff
@@ -0,0 +1,159 @@
+#include "window.hpp"
+#include "config.hpp"
+#include "drawable.hpp"
+#include <vector>
+#include <xcb/xproto.h>
+
+/* NOTE: First part is mostly close to DesktopPixmap */
+window::window (const int16_t x,         ///< x coordinate of the top left corner
+                const int16_t y,         ///< y coordinate of the top left corner
+                const uint16_t width,    ///< width of display
+                const uint16_t height,   ///< height of display
+                const std::string &name, ///< _NET_WM_NAME of display
+                const uint32_t parent)
+    : drawable (x, y, width, height)
+{
+  this->name = name;
+  this->b_width = 0;
+  this->parent = parent;
+
+  // Connection to X-server, as well as generating id before drawing window
+  // itself
+  // Sets default parent to root
+  if (this->parent == XCB_NONE)
+    {
+      this->id = xcb_generate_id (drawable::c_);
+      this->parent = drawable::screen_->root;
+      create_window ();
+    }
```

Comment:
This is redundant. Remove `this->parent` and explicitly use `drawable::screen_->root` or add `this->root` variable